lib/cli/run-helpers.js

```diff
@@ -92,7 +92,10 @@ exports.handleRequires = async (requires = []) => {
       debug('resolved required file %s to %s', mod, modpath);
     }
     const requiredModule = await requireOrImport(modpath);
-    if (type(requiredModule) === 'object' && requiredModule.mochaHooks) {
+    if (
+      ['object', 'module'].includes(type(requiredModule)) &&
```

Comment:
```node -e "import('./test/integration/fixtures/options/require/root-hook-defs-esm.fixture.mjs').then(mod => console.log(Object.prototype.toString.call(mod)))"```
-> `[object Module]`

Piped through the `type` function: https://github.com/mochajs/mocha/blob/master/lib/utils.js#L216
Returns `module`.

So added handling for that "module" case (which seems reasonable considering that is what we would expect from ESM... but for the sake of discussing alternatives, could add some handling to `type()` to map module -> object

Reply:
no, this seems correct.  but, we should also add a test in `test/node-unit/utils.spec.js` to cover the behavior.  you don't need to do this unless you want to.

Reply:
we can change it to return `object` if there are unforeseen consequences.  what does `typeof requiredModule` return?

Reply:
```node -e "import('./test/integration/fixtures/options/require/root-hook-defs-esm.fixture.mjs').then(mod => console.log(typeof mod))"``` -> `object`

I'll let you interpret that... since there must be _some_ reason not to use `typeof` directly.

> add a test to cover the behavior
I can add that as well, so long as we are fine with Modules returning "module"

Reply:
there's no real great reason to use `type()` here other than trying to maintain some consistency. we should use `typeof` and not worry about it.

Reply:
Ref: #4306